File: CS50x 2023 Solutions/Week 1/ProbSet 1/cash.c
#include "cs50.h"
#include <stdio.h>

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int quarters;
    for (quarters = 0; cents >= 25; quarters++)
    {
        cents -= 25;
    }

    int dimes;
    for (dimes = 0; cents >= 10; dimes++)
    {
        cents -= 10;
    }

    int nickels;
    for (nickels = 0; cents >= 5; nickels++)
    {
        cents -= 5;
    }

    int pennies;
    for (pennies = 0; cents > 0; pennies++)
    {
        cents -= 1;
    }

    int coins = quarters + dimes + nickels + pennies;

    printf("%i\n", coins);
}

File: CS50x 2023 Solutions/Week 6/ProbSet 6/cash.py
from cs50 import get_float


while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

change = int(change * 100)
total = 0
while change >= 25:
    change -= 25
    total += 1
while change >= 10:
    change -= 10
    total += 1
while change >= 5:
    change -= 5
    total += 1
while change >= 1:
    change -= 1
    total += 1

print(f"{total}")
